newIDE/app/src/MainFrame/Preferences/PreferencesProvider.js

```diff
@@ -343,6 +346,46 @@ export default class PreferencesProvider extends React.Component<Props, State> {
     );
   }
 
+  _getRecentProjectFiles() {
+    return this.state.values.recentProjectFiles;
+  }
+
+  _setRecentProjectFiles(recents: Array<FileMetadata>) {
+    this.setState(
+      state => ({
+        values: {
+          ...state.values,
+          recentProjectFiles: recents,
+        }
+      }),
+      () => this._persistValuesToLocalStorage(this.state)
+    );
+  }
```

Comment:
I decided not to expose this setter function, because insertion in recent files has a particular method (which is sort of LIFO, but not exactly, meaning if a file already exists in the array, it should come on top).

Reply:
Yes it's fine, the context should only expose functions that are safe and straightforward to use like `_insertRecentProjectFile`. The setter is too powerful to put it in the end of the components, they simply don't need it ;)